File: Array/intersection.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
         boolean[] used = new boolean[nums2.length];
        int[] temp = new int[Math.min(nums1.length, nums2.length)];
        int index = 0;

        for (int i = 0; i < nums1.length; i++) {
            for (int j = 0; j < nums2.length; j++) {
                if (!used[j] && nums1[i] == nums2[j]) {
                    temp[index++] = nums1[i];
                    used[j] = true;
                    break;
                }
            }
        }

        int[] result = new int[index];
        for (int i = 0; i < index; i++) {
            result[i] = temp[i];
        }
        return result;
    }

File: Array/intersection.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = []
        nums2_used = [False] * len(nums2)

        for num in nums1:
            for i in range(len(nums2)):
                if not nums2_used[i] and nums2[i] == num:
                    result.append(num)
                    nums2_used[i] = True
                    break

        return result
